crates/foundry/evm/core/src/opts.rs

```diff
@@ -1,22 +1,28 @@
+use std::marker::PhantomData;
+
 use alloy_chains::Chain;
 use alloy_network::AnyRpcBlock;
 use alloy_primitives::{Address, B256, U256};
 use alloy_provider::Provider;
 use eyre::WrapErr;
-use revm::primitives::{BlockEnv, CfgEnv, TxEnv};
+use revm::{
+    context::{BlockEnv, CfgEnv, TxEnv},
+    primitives::hardfork::SpecId,
+};
 use serde::{Deserialize, Deserializer, Serialize};
 use url::Url;
 
 use super::fork::{environment, provider::ProviderBuilder};
+use crate::evm_env::{BlockEnvTr, EvmEnv, HardforkTr, TransactionEnvTr};
 
 #[derive(Clone, Debug, Default, Serialize, Deserialize)]
-pub struct EvmOpts {
+pub struct EvmOpts<BlockT, TxT, HardforkT> {
     /// The EVM environment configuration.
     #[serde(flatten)]
     pub env: Env,
 
     /// The hardfork to use for the EVM.
-    pub spec: revm::primitives::SpecId,
+    pub spec: SpecId,
```

Comment:
What's the reason this is a `SpecId` rather than a `HardforkT`?

Reply:
Thanks, it should be `HardforkT`. Fixed in https://github.com/NomicFoundation/edr/pull/901/commits/e4571e01e3f597bf60cda508725f73a18b93f107